test/test_variables.py

```diff
@@ -115,6 +115,27 @@ def update_pass(prefix):
         update_pass("applications")
         assert self.get(url='/3')['status'] == 404
 
+    def test_variables_dollar(self):
```

Comment:
Please apply the following style fixes for the tests (nothing significant, just making life easier if we decide to add more tests in the future).

Also I would suggest to move this test a little bit closer to the similar test for $host, $method, etc (to match organization in C code). This is not a requirement.

```
diff --git a/test/test_variables.py b/test/test_variables.py
--- a/test/test_variables.py
+++ b/test/test_variables.py
@@ -53,6 +53,27 @@ class TestVariables(TestApplicationProto
         check_host('www.localhost', 404)
         check_host('localhost1', 404)
 
+    def test_variables_dollar(self):
+        assert 'success' in self.conf(
+            {
+                "listeners": {"*:7080": {"pass": "routes"}},
+                "routes": [{"action": {"return": 301}}],
+            }
+        )
+
+        def check_dollar(location, expect):
+            assert 'success' in self.conf(
+                '"' + location + '"',
+                'routes/0/action/location',
+            )
+            assert self.get()['headers']['Location'] == expect
+
+        check_dollar(
+            'https://${host}${uri}path${dollar}dollar',
+            'https://localhost/path$dollar',
+        )
+        check_dollar('path$dollar${dollar}', 'path$$')
+
     def test_variables_many(self):
         self.conf_routes("\"routes$uri$method\"")
         assert self.get(url='/5')['status'] == 206, 'many'
@@ -115,27 +136,6 @@ class TestVariables(TestApplicationProto
         update_pass("applications")
         assert self.get(url='/3')['status'] == 404
 
-    def test_variables_dollar(self):
-        assert 'success' in self.conf(
-            {
-                "listeners": {"*:7080": {"pass": "routes"}},
-                "routes": [{"action": {"return": 301}}],
-            }
-        )
-
-        assert 'success' in self.conf(
-            '"https://${host}${uri}path${dollar}sub"',
-            'routes/0/action/location'
-        ), 'location with $'
-        assert self.get()['headers']['Location'] == 'https://localhost/path$sub'
-
-        assert 'success' in self.conf(
-            '"path$dollar${dollar}"', 'routes/0/action/location'
-        ), 'path followed by $$'
-        assert (
-            self.get()['headers']['Location'] == 'path$$'
-        )
-
     def test_variables_invalid(self):
         def check_variables(routes):
             assert 'error' in self.conf(

```

Reply:
Ah, OK I see. Thanks!